controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -195,4 +297,25 @@ public void failed(State from, Throwable failure) {
     Map<Integer, BucketService> getBucketServices() {
         return Collections.unmodifiableMap(buckets);
     }
+
+    /**
+     * Method to start all listener required for bucket manager.
+     */
+    private void startAllListeners() {
+        startBucketControllerMapListener();
+        startLeaderElection();
+        startBucketOwnershipListener();
+    }
+
+    /**
+     * Start a new bucket service on controller instance. If any exception occurs while starting the bucket, it will
+     * retry for 10 times.
+     *
+     * @param bucketId Id of bucket service.
+     * @return future, which when complete indicate that ownership acquired successfully.
+     */
+    protected CompletableFuture<Void> startBucketService(int bucketId) {
+       return RetryHelper.withRetriesAsync(() -> tryTakeOwnership(bucketId),
+               RetryHelper.UNCONDITIONAL_PREDICATE, NO_OF_RETRIES, getExecutor());
```

Comment:
If retries exhausted, it will simply stop retrying. It will try to acquire it again either existing owner owner release ownership later(If it was unable to do that) or redistribution of buckets happens.
This [test](https://github.com/pravega/pravega/blob/dab2e15f251673264e92c6f004a2e7fa91ca51cb/controller/src/test/java/io/pravega/controller/server/bucket/PravegaTablesStoreBucketServiceTest.java#L107) is verifying this functionality.